src/stream/stream/peekable.rs

```diff
@@ -0,0 +1,65 @@
+use std::pin::Pin;
+
+use crate::future::Future;
+use crate::stream::Stream;
+use crate::task::{Context, Poll};
+
+#[doc(hidden)]
+#[allow(missing_debug_implementations)]
+pub struct Peekable<S: Stream>
+where
+    S: Sized,
+{
+    stream: S,
+    peeked: Option<Option<S::Item>>,
+}
+
+//pub struct PeekFuture<'a, T: Unpin + ?Sized> {
+//pub(crate) stream: &'a T,
+//}
+
+//impl<T: Stream + Unpin + ?Sized> Future for PeekFuture<'_, T> {
+//}
+
+impl<S> Peekable<S>
+where
+    S: Stream,
+{
+    pin_utils::unsafe_pinned!(stream: S);
+    pin_utils::unsafe_unpinned!(peeked: Option<Option<S::Item>>);
```

Comment:
It'd be nice if this could be updated to use `pin-project-lite` the way we've done for other combinators (we merged this in the last week; in general it should be nicer!)